### ISSUE TITLE
False positive SA4010: this result of append is never used, except maybe in other appends 

### ISSUE THREAD

**Issue by WtfJoke** [golangci/golangci-lint#624]
Hi there,

Im quite new to golang, but think this linter check is a false positive in my case:
```go
package main

func something() {
	var translations = map[string]string{
		"a": "translation",
		"b": "other"}
	names := make([]string, 0, len(translations))
	for _, v := range translations {
		names = append(names, v)
	}
}
```

If you need something else or think it isnt a false positive, feel free to close it 😄 

Please include the following information:
1. Version of golangci-lint: `golangci-lint --version` (or git commit if you don't use binary distribution)
v1.17.1
2. Config file: `cat .golangci.yml`
No config file
3. Go environment: `go version && go env`
```
go version go1.12.7 linux/amd64
GOARCH="amd64"
GOBIN=""
GOCACHE="/home/developer/.cache/go-build"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOOS="linux"
GOPATH="/home/developer/go"
GOPROXY=""
GORACE=""
GOROOT="/usr/local/go"
GOTMPDIR=""
GOTOOLDIR="/usr/local/go/pkg/tool/linux_amd64"
GCCGO="gccgo"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD=""
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build796517251=/tmp/go-build -gno-record-gcc-switches"
```
4. Verbose output of running: `golangci-lint run -v`

```
golangci-lint run -v           
INFO [config_reader] Config search paths: [./ /home/developer/go/src/sample-go /home/developer/go/src /home/developer/go /home/developer /home /] 
INFO [lintersdb] Active 10 linters: [deadcode errcheck gosimple govet ineffassign staticcheck structcheck typecheck unused varcheck] 
INFO [lintersdb] Optimized sublinters [staticcheck gosimple unused] into metalinter megacheck 
INFO [loader] Go packages loading at mode load types and syntax took 2.284056637s 
INFO [loader] SSA repr building timing: packages building 17.187591ms, total 133.279459ms 
INFO [runner] worker.3 took 1.995337ms with stages: varcheck: 677.819µs, errcheck: 518.515µs, deadcode: 437.037µs, structcheck: 349.431µs, typecheck: 2.375µs 
INFO [runner] worker.2 took 11.600944ms with stages: ineffassign: 11.596534ms 
INFO [runner] worker.4 took 381.680083ms with stages: govet: 381.672601ms 
INFO [runner] worker.1 took 1.949815581s with stages: megacheck: 1.949803381s 
INFO [runner] Workers idle times: #2: 1.9381148s, #3: 1.947662539s, #4: 1.568023177s 
INFO [runner] Issues before processing: 4, after processing: 1 
INFO [runner] processing took 1.433963ms with stages: nolint: 700.761µs, exclude: 273.451µs, identifier_marker: 170.399µs, autogenerated_exclude: 169.225µs, source_code: 29.687µs, skip_dirs: 28.224µs, cgo: 27.71µs, path_prettifier: 21.998µs, uniq_by_line: 3.405µs, filename_unadjuster: 2.24µs, max_same_issues: 2.005µs, path_shortener: 1.231µs, max_from_linter: 1.117µs, max_per_file_from_linter: 890ns, diff: 793ns, exclude-rules: 454ns, skip_files: 373ns 
test.go:9:17: SA4010: this result of append is never used, except maybe in other appends (staticcheck)
                names = append(names, v)
                              ^
INFO File cache stats: 1 entries of total size 223B 
INFO Memory: 40 samples, avg is 438.2MB, max is 876.4MB 
INFO Execution took 4.423052881s                  
```

**Comment by alexkohler** [comment]
I would recommend filing an issue with `staticcheck` itself rather than golangci: https://github.com/dominikh/go-tools

**Comment by WtfJoke** [comment]
Closing this in favor of https://github.com/dominikh/go-tools/issues/544
